pkg/deploy/controller.go

```diff
@@ -251,6 +251,17 @@ func checksum(bytes []byte) string {
 	return hex.EncodeToString(d[:])
 }
 
+func isEmptyYaml(yaml []byte) bool {
+	isEmpty := true
+	lines := bytes.Split(yaml, []byte("\n"))
+	for _, k := range lines {
+		if string(k) != "---" && !bytes.HasPrefix(k, []byte("#")) && string(k) != "" {
```

Comment:
What about a line with just spaces/tabs? Or a line with a couple spaces followed by a `#`?

Reply:
good point. I've added whitespace trimming, should work for such cases. 
It'd be really nice to have some integration tests in k3s CI scripts...